### PR TITLE
CIRCSTORE-99 Add tags field to Request DTO

### PR DESCRIPTION
**Pull request by Anatolii-Starkov**
Add tags field to the Request DTO.
To display, assign, unassign, and output tags associated with
individual REQUEST records within the FOLIO Requests app.

### CODE REVIEW

File: src/test/java/org/folio/rest/support/builders/RequestRequestBuilder.java [hunk 1]
```diff
@@ -718,7 +750,33 @@ public RequestRequestBuilder withPickupServicePointId(UUID pickupServicePointId)
...
+      this.itemSummary,
+      this.requesterSummary,
+      this.proxySummary,
+      this.status,
+      this.cancellationReasonId,
+      this.cancelledByUserId,
+      this.cancellationAdditionalInformation,
+      this.cancelledDate,
+      this.position,
+      pickupServicePointId,
```
Comment (mreno-EBSCO) [inline src/test/java/org/folio/rest/support/builders/RequestRequestBuilder.java]: As mentioned above, to make this consistent, change `pickupServicePointId` to `this.pickupServicePointId`.

File: src/test/java/org/folio/rest/support/builders/RequestRequestBuilder.java [hunk 2]
```diff
@@ -180,7 +191,8 @@ public RequestRequestBuilder withRequestType(String requestType) {
       this.cancellationAdditionalInformation,
       this.cancelledDate,
       this.position,
-      this.pickupServicePointId);
+      this.pickupServicePointId,
+      tags);
```
Comment (mreno-EBSCO) [inline src/test/java/org/folio/rest/support/builders/RequestRequestBuilder.java]: To make this consistent, change `tags` to `this.tags`. This is the case for all the `withXXX()` methods in this file, except `withTags()`.

File: src/test/java/org/folio/rest/api/RequestsApiTest.java [hunk 3]
```diff
@@ -9,9 +10,9 @@
 import static org.folio.rest.support.matchers.TextDateTimeMatcher.withinSecondsAfter;
 import static org.folio.rest.support.matchers.ValidationErrorMatchers.hasMessage;
 import static org.folio.rest.support.matchers.ValidationResponseMatchers.isValidationResponseWhich;
-import static org.hamcrest.CoreMatchers.containsString;
-import static org.hamcrest.CoreMatchers.not;
+import static org.hamcrest.CoreMatchers.*;
 import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.collection.IsCollectionWithSize.hasSize;
```
Comment (mreno-EBSCO) [inline src/test/java/org/folio/rest/api/RequestsApiTest.java]: Remove unused import

File: descriptors/ModuleDescriptor-template.json [hunk 4]
```diff
@@ -89,7 +89,7 @@
     },
     {
       "id": "request-storage",
-      "version": "2.4",
+      "version": "2.5",
```
Comment (marcjohnson-kint) [inline descriptors/ModuleDescriptor-template.json]: This now needs to be `2.6` as [recent changes](https://github.com/folio-org/mod-circulation-storage/commit/6ae2901d2d78387b8c5a1a9b332bc5509ce2372f) already published `2.5`

Also, please update the version in the RAML documentation as well - https://github.com/folio-org/mod-circulation-storage/blob/2695111751afd5e153db60ff7f1ff2cb7c10aa05/ramls/request-storage.raml#L3

File: src/test/java/org/folio/rest/api/RequestsApiTest.java [hunk 5]
```diff
@@ -156,6 +159,13 @@ public void canCreateARequest()
 
     assertThat("barcode is taken from proxying user",
       proxyRepresentation.getString("barcode"), is("6059539205"));
+
+    assertThat(representation.containsKey("tags"), is(true));
+
+    final JsonObject tagsRepresentation = representation.getJsonObject("tags");
+
+    assertThat(tagsRepresentation.containsKey("tagList"), is(true));
+    assertThat(tagsRepresentation.getJsonArray("tagList"), contains("new", "important"));
```
Comment (marcjohnson-kint) [inline src/test/java/org/folio/rest/api/RequestsApiTest.java]: Thanks for adding an example to the tests

**Pull request merged**